enhancements/console/workflow-guides.md

```diff
@@ -0,0 +1,175 @@
+2. Make users understand what are the steps needed to achieve their goals.
+3. Have a uniform format for writing the Workflow Guides.
+4. Have a storage for the Workflow Guides.
+5. Easy access to the Workflow Guides outside of OpenShift Console for all sorts of purposes(docs, forums, blogs, demos, ...).
+6. **Extenstion:** Show user list of resources that need to be created in order to complete the workflow.
+
+### Non-Goals
+
+1. Provide API for writing Workflow Guides.
+2. Provide new format for writing the Workflow Guides.
+3. Provide a dedicated storage for the Workflow Guides.
+4. Lock the Workflow Guides inside the OpenShift Console.
+
+## Proposal
+
+* Introduce format that will be used for writing Workflow Guides.
+* Introduce default storage for Workflow Guides.
+* Introduce mechanism how to connect different parts of OpenShift Console (Operators, Workloads, ...) with the Workflow Guides.
+* **Extenstion:**: Introduce mechanism for defining and displaying list of required resources, that need to be created in order to complete the Workflow Guide.
+
+### User Stories
+
+#### Story 1
+
+As an administrator of an OpenShift cluster, I need a guide to walk me through how to install OpenShift Serverless modules (Serving and Eventing) in a cluster.
+
+#### Story 2
+
+As an administrator of an OpenShift cluster, I need a guide to walk me through how to update an OpenShift cluster.
+
+#### Story 3
+
+As a developer, I need a guide to walk me through how to deploy an existing application as a serverless workload.
+
+#### Story 4
+
+As a operator creator I want to provide operator consumers with a guide on how to install and user the my operator.
+
+#### Story 5
+
+As a OpenShift evangelist I want to link developers on StackOverflow with complete Workflow Guides, for their issues.
+
+### Implementation Details
+
+1. GitHub will be used creating and storing the Workflow Guides. GitHub provides rich markdown options, including images and videos, which will come handy in the process of creating a Workflow Guide. Since we have storage limits in ETCD, it's for the best to use a technology that both storest the content and provides rich markdown. Also by using GitHub we solve the issue with  Workflow Guides versioning.
+2. For linking the OpenShift Console and the Workflow Guides `ConsoleLinks` CRD will be used. To suite the need following extension of the `ConsoleLinks` is needed:
+   1. Add additional `ConsoleLinkLocation` type to the `ConsoleLink` [API](https://github.com/openshift/api/blob/master/console/v1/types_console_link.go#L65-L74) named `WorkflowGuide`.
+   2. If `ConsoleLinkLocation` type is set to `WorkflowGuide`, it will be possible to define a `WorkflowGuideSpec` with following fields:
+      * Section - string - Defines area of interest, e.g. `operators`.
+      * Admin - bool - Defines if the Workflow Guide is cluster admin oriented (user has to have cluster admin permisions).
```

Comment:
Thinking about this now it might be a good idea to be have the link agnostic on the user permissions. Since the links will be rendered in:
- element that represents start of given workflow (either what [UX team is proposing](https://marvelapp.com/236ge4ig/screen/69913818) or what [I was](https://raw.githubusercontent.com/jhadvig/images/master/operators.png))
- list of Guides

It shouldn't really matter I think since if user sees the element for starting a particular workflow then he will also see the link for the tour. Otherwise he would only see it in the list of all the Guides. 
The `Admin` field was just for filtering purposes really. meaning which Guides we wanna show when user lists the Guides Dev Console and which for Admin Console, to only show relevant Guides.

Dont think that rendering links for the Guides, either linking to our docs or GH should cause any security issues